kronofoto/static/assets/js/lib.js

```diff
@@ -18,633 +18,808 @@ import { Tooltip } from 'foundation-sites/js/foundation.tooltip';
+      self.jquery('.collection-name').text(self.dragOffset)
+      newIndex = dragOffsetIndex + ((self.dragOffset - deltaX) != 0 ? Math.round((self.dragOffset - deltaX) / thumbnailWidth) : 0) + 1 //dragOffsetIndex - Math.round((self.dragOffset + deltaX) / thumbnailWidth) + 1
+    }
+    let newItem = self.jquery('#fi-thumbnail-carousel-images li:nth-child('+newIndex+')')
+    let contextWidth = self.root.offsetWidth
+
+    let numPrev = self.jquery(newItem).prevAll().length
+    let desiredThumbBuffer = 0.75 * contextWidth / thumbnailWidth
+    let numNext = self.jquery(newItem).nextAll().length
+    if (numPrev < desiredThumbBuffer) {
+      self.getMoreThumbnailsLeft()
+    } else if (numNext < desiredThumbBuffer) {
+      self.getMoreThumbnailsRight()
+    }
+
+    self.jquery('#fi-thumbnail-carousel-images li').removeAttr('data-active')
+    self.jquery(newItem).attr('data-active', '')
+  }
+
+  getMoreThumbnailsRight() {
+    let self = this
+    if(!self.gettingMoreThumbnailsRight) {
+      self.gettingMoreThumbnailsRight = true
+      let form = self.jquery('#thumbnail-request').get(0)
+      let formData = new FormData(form)
+      let targetId = self.jquery('#fi-thumbnail-carousel-images li:last-child span').attr('hx-get')
+      targetId = parseInt(targetId.split('/')[targetId.split('/').length-1].replace(/[^0-9]/g, ''))
+      formData.set('id', targetId)
+      formData.set('forward', 'True')
+      var headers = {
+        'Hx-Request': true,
+        'Hx-Target': 'thumbnail-request',
+        'Hx-Trigger': 'thumbnail-request'
+      }
+      let action = form.getAttribute('hx-get') + '?' + (new URLSearchParams(formData))
+      fetch(action, {
+        method: 'GET',
+        headers: headers,
+      })
+        .then(response => response.text())
+        .then(data => {
+          // Request was successful, handle the response here
+          self.gettingMoreThumbnailsRight = false
+          self.jquery('#fi-thumbnail-carousel-images').append(data)
+          self.htmx.process(self.jquery('#fi-thumbnail-carousel-images').get(0))
         })
-        $('#fi-thumbnail-carousel-images').prepend($before)
-        htmx.process($('#fi-thumbnail-carousel-images').get(0))
+        .catch(error => {
+          // Handle any errors that occurred during the fetch
+          console.error('Error:', error)
+        });
+    }
+  }
+
+  getMoreThumbnailsLeft() {
+    let self = this
+    if(!self.gettingMoreThumbnailsLeft) {
+      self.gettingMoreThumbnailsLeft = true
+      let form = self.jquery('#thumbnail-request').get(0)
+      let formData = new FormData(form)
+      let targetId = self.jquery('#fi-thumbnail-carousel-images li:first-child span').attr('hx-get')
+      targetId = parseInt(targetId.split('/')[targetId.split('/').length-1].replace(/[^0-9]/g, ''))
+      formData.set('id', targetId)
+      formData.set('forward', 'False')
+      var headers = {
```

Comment:
I think this is pretty clunky. My intention was to use elem.closest("form") to get the parent form instance, change the id and forward as needed, and hx.trigger(elem, "kronofoto:loadThumbnails"). It's not implemented yet, but the response can trigger an event on an element with whatever data you need. Or the swap could put the thumbnails in a hidden div and the event handler could look for them there.